runtime/pangolin-parachain/src/pallets/fee_market.rs

```diff
@@ -0,0 +1,55 @@
+// --- core ---
+use core::cmp;
+// --- substrate ---
+use frame_support::{traits::LockIdentifier, PalletId};
+use sp_runtime::{traits::UniqueSaturatedInto, Permill};
+// --- darwinia ---
+use crate::*;
+use pallet_fee_market::{Config, RingBalance, Slasher};
+
+pub struct FeeMarketSlasher;
+impl<T: Config> Slasher<T> for FeeMarketSlasher {
+	fn slash(locked_collateral: RingBalance<T>, timeout: T::BlockNumber) -> RingBalance<T> {
+		let slash_each_block = 2 * COIN;
+		let slash_value = UniqueSaturatedInto::<Balance>::unique_saturated_into(timeout)
+			.saturating_mul(UniqueSaturatedInto::<Balance>::unique_saturated_into(
+				slash_each_block,
+			))
+			.unique_saturated_into();
+
+		cmp::min(locked_collateral, slash_value)
+	}
+}
+
+frame_support::parameter_types! {
+	pub const FeeMarketPalletId: PalletId = PalletId(*b"da/feemk");
+	pub const TreasuryPalletId: PalletId = PalletId(*b"da/trsry");
```

Comment:
IIUC, fee market doesn't need treasury pallet. Just an account here.

Reply:
fee market uses this id to generate the treasury account_id. Do you mean hardcode the treasury account_id in the fee market runtime config directly?

Reply:
It's okay without treasury pallet.
Treasury account still will be created based on the id.